### ISSUE TITLE
GFWlist proxy doesn't work after upgrading to new version

### ISSUE THREAD

**Issue by RoninYuan-WY** [kuoruan/luci-app-v2ray#61]
Can't access any website since choosing GFWList Proxy, even sometimes can't update V2ray setting in router WebUI

**Comment by RoninYuan-WY** [comment]
I think it broke the GFWList file. it works after updating.

**Comment by RoninYuan-WY** [comment]
I found after opening several websites, something else was broken. connection down

**Comment by RoninYuan-WY** [comment]
I found after opening several websites, something else was broken. connection down

**Comment by kuoruan** [comment]
There should be some ip in ipset proxy list.
```
ipset list v2ray_dst_proxy_v4
```

Check your dnsmasq service.

```
ps -w | grep dnsmasq
cat /tmp/dnsmasq.d/v2ray | head -n5
```

**Comment by RoninYuan-WY** [comment]
catch that when switch to GFWlist proxy?

**Comment by RoninYuan-WY** [comment]
root@OpenWrt:~# ps -w | grep dnsmasq
25904 dnsmasq  23180 S    /usr/sbin/dnsmasq -C /var/etc/dnsmasq.conf.cfg01411c -k -x /var/run/dnsmasq/dnsmasq.cfg01411c.pid
25968 root      3092 S    grep dnsmasq
root@OpenWrt:~# cat /tmp/dnsmasq.d/v2ray | head -n5
# AUTO-GENERATED FILE. DO NOT MODIFY.
server=/nvidia.com/114.114.114.114#53
ipset=/nvidia.com/v2ray_dst_direct_v4,v2ray_dst_direct_v6
server=/linkedin.com/114.114.114.114#53
ipset=/linkedin.com/v2ray_dst_direct_v4,v2ray_dst_direct_v6

**Comment by RoninYuan-WY** [comment]
strange thing is that youtube is fine but other Websites in GFWlist were down.

**Comment by kuoruan** [comment]
Is there any ip in your ipset proxy list?

```
resolveip -4 -t 5 www.google.com
```
Get the ip and try:
```
ipset list v2ray_dst_proxy_v4 | grep OUTPUT_IP
```

**Comment by RoninYuan-WY** [comment]
root@OpenWrt:~# resolveip -4 -t 5 www.google.com
69.171.242.11
root@OpenWrt:~# ipset list v2ray_dst_proxy_v4 | grep 69.171.242.11
root@OpenWrt:~# 

It seems empty

**Comment by kuoruan** [comment]
Any ips in your proxy list?

```
ipset list v2ray_dst_proxy_v4
```

**Comment by kuoruan** [comment]
My output is like:
```
root@OpenWrt:~# resolveip www.google.com
2404:6800:4008:802::2004
31.13.83.16
root@OpenWrt:~# ipset list v2ray_dst_proxy_v4
Name: v2ray_dst_proxy_v4
Type: hash:net
Revision: 6
Header: family inet hashsize 64 maxelem 65536 timeout 604800
Size in memory: 816
References: 1
Number of entries: 7
Members:
172.217.24.14 timeout 604741
13.250.177.223 timeout 604742
173.252.103.64 timeout 604678
52.74.223.119 timeout 604769
216.58.200.46 timeout 604768
31.13.68.22 timeout 604639
31.13.83.16 timeout 604680
```

you can see the resolve ip is in the proxy list

**Comment by kuoruan** [comment]
If not, confirm you are installing dnsmasq-full and your dnsmasq supports ipset.

**Comment by RoninYuan-WY** [comment]
I installed dnsmasq-full 2.80-1.4

**Comment by RoninYuan-WY** [comment]
I reset the router and start from the scratch, install v2ray core and your release, then configured. and got same result.

**Comment by RoninYuan-WY** [comment]
I used another vps and different ws+tls transparent settings, I saw same problem, I think maybe it's related to GFWlist settings as well as dns proxy stuff.

**Comment by kuoruan** [comment]
Check your system log to see if there is error.

**Comment by RoninYuan-WY** [comment]
Wed Dec  4 12:43:08 2019 daemon.info v2ray: Policy disabled: main_policy
Wed Dec  4 12:43:08 2019 daemon.info v2ray: Reverse disabled: main_reverse
Wed Dec  4 12:43:08 2019 daemon.info v2ray: Setting transparent proxy on port: 1081
Wed Dec  4 12:43:08 2019 daemon.info v2ray: Transparent proxy mode: gfwlist_proxy
Wed Dec  4 12:43:08 2019 daemon.info v2ray: Restarting dnsmasq...
Wed Dec  4 12:43:08 2019 daemon.info dnsmasq[4357]: exiting on receipt of SIGTERM
Wed Dec  4 12:43:08 2019 user.notice dnsmasq: DNS rebinding protection is active, will discard upstream RFC1918 responses!
Wed Dec  4 12:43:08 2019 user.notice dnsmasq: Allowing 127.0.0.0/8 responses
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: started, version 2.80 cachesize 150
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: DNS service limited to local subnets
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: compile time options: IPv6 GNU-getopt no-DBus no-i18n no-IDN DHCP DHCPv6 no-Lua TFTP conntrack ipset auth DNSSEC no-ID loop-detect inotify dumpfile
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq-dhcp[4630]: DHCP, IP range 192.168.2.100 -- 192.168.2.249, lease time 12h
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using local addresses only for domain test
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using local addresses only for domain onion
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using local addresses only for domain localhost
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using local addresses only for domain local
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using local addresses only for domain invalid
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using local addresses only for domain bind
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zzux.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zzcloud.me
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zzcartoon.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zyzc9.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zyns.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zynamics.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zynaima.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zvereff.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zuola.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zuobiao.me
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zuo.la
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zsrhao.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zspeeder.me
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zpn.im
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zozotown.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zorrovpn.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zoozle.net
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zootool.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zoogvpn.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zonghexinwen.net
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zonghexinwen.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zonaeuropa.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zomobo.net
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zodgame.us
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zmw.cn
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zkaip.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zippyshare.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain ziporn.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zinio.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zim.vn
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using 5470 more nameservers
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: reading /tmp/resolv.conf.auto
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using local addresses only for domain test
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using local addresses only for domain onion
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using local addresses only for domain localhost
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using local addresses only for domain local
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using local addresses only for domain invalid
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using local addresses only for domain bind
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zzux.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zzcloud.me
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zzcartoon.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zyzc9.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zyns.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zynamics.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zynaima.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zvereff.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zuola.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zuobiao.me
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zuo.la
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zsrhao.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zspeeder.me
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zpn.im
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zozotown.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zorrovpn.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zoozle.net
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zootool.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zoogvpn.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zonghexinwen.net
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zonghexinwen.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zonaeuropa.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zomobo.net
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zodgame.us
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zmw.cn
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zkaip.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zippyshare.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain ziporn.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zinio.com
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using nameserver 8.8.8.8#53 for domain zim.vn
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: using 5474 more nameservers
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: read /etc/hosts - 4 addresses
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: read /tmp/hosts/odhcpd - 3 addresses
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq[4630]: read /tmp/hosts/dhcp.cfg01411c - 2 addresses
Wed Dec  4 12:43:11 2019 daemon.info dnsmasq-dhcp[4630]: read /etc/ethers - 0 addresses

**Comment by RoninYuan-WY** [comment]
I found DHCP-Options, 6,1.1.1.1,114.114.114.114 caused dnsmasq fail to start. change to 6,1.1.1.1 will fix problem, strange. 
Another issue, GFWlist now works for PC, but have problem with huge packets lost for the phone with wireless access to router(almost can't open anything behind FW), but other options like CN direct, default etc won't meet such an issue.

**Comment by RoninYuan-WY** [comment]
just give you an idea, I belive that GFWlist has some problem with V2ray itself. I install lede with other v2ray extension on a new router, found that the phone using WIFI has apps  behind GFW like youtube not accessible and browser on the phone can access those website intermittently. the behaviour is just like your app, switching to CN direct will work fine.

**Comment by kuoruan** [comment]
OK, got it.

**Comment by kuoruan** [comment]
Fixed the DNS proxy in 1.5.1, have a try.
